src/cobalt_model/pagination_config.rs

```diff
@@ -1,108 +1,109 @@
-use liquid;
-use std::borrow::Cow;
+use std::convert::Into;
 use std::vec::Vec;
+
+use liquid;
+
 use super::SortOrder;
 
-// struct Trails {
-//   before: i32,
-//   after: i32,
-// }
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub enum Include {
+  None,
+  #[serde(rename = "all")]
+  All,
+  // tags,
+}
+
+impl Into<liquid::Scalar> for Include {
```

Comment:
I'd personally recommend removing the `liquid` policy from this and have it just return `&'static str`.

Reply:
I've done this:
```rust
impl Into<&'static str> for Include {
  fn into(self) -> &'static str {
    match self {
      Include::None => "",
      Include::All => "all",
    }
  }
}
```

but with:
```rust
attributes.insert("include".to_owned(), liquid::Value::scalar(include.into()));
```

I got this error:
```rust
error[E0283]: type annotations required: cannot resolve `_: std::convert::Into<liquid::Scalar>`
   --> src\pagination.rs:236:43
    |
236 |   attributes.insert("include".to_owned(), liquid::Value::scalar(include.into()));
    |                                           ^^^^^^^^^^^^^^^^^^^^^
    |
    = note: required by `liquid::Value::scalar`
```

I had to do this to make it compiles:
```rust
let i: &str = include.into();
attributes.insert("include".to_owned(), liquid::Value::scalar(i));
```

I tried `liquid::Value::scalar(include.into() as &str)` with no luck.

Is there a trick to avoid this ugly extra line just to have a type annotation?

Reply:
This error is a bit misleading.

I think the problem is the fact that we have
- `Into<&str> for Include`
- `Into<Scalar> for &str`

It can't know to call `into` twice to convert `Include` into `&str` into `Scalar`.

You could simplify it to
```rust
let i: &str = include.into();
attributes.insert("include".to_owned(), i.into());
```
It should know that the value being inserted is a `Scalar` so it should be able to call the into correctly.

Reply:
It seems that `liquid::Value::scalar(i)` is mandatory. With `i.into()`:
```rust
error[E0277]: the trait bound `liquid::Value: std::convert::From<&str>` is not satisfied
   --> src\pagination.rs:233:45
    |
233 |   attributes.insert("include".to_owned(), i.into());
    |                                             ^^^^ the trait `std::convert::From<&str>` is not implemented for `liquid::Value`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<liquid::Value>` for `&str`
```